src/clients/cli.rs

```diff
@@ -407,79 +405,64 @@ mod tests {
             vec![WaitFor::message_on_stdout("Hello from Docker!")]
         }
 
-        fn args(&self) -> <Self as Image>::Args {
-            vec![]
-        }
-
         fn env_vars(&self) -> Box<dyn Iterator<Item = (&String, &String)> + '_> {
             Box::new(self.env_vars.iter())
         }
 
         fn volumes(&self) -> Box<dyn Iterator<Item = (&String, &String)> + '_> {
             Box::new(self.volumes.iter())
         }
-
-        fn with_args(self, _arguments: <Self as Image>::Args) -> Self {
-            self
-        }
     }
 
     #[test]
     fn cli_run_command_should_include_env_vars() {
-        let mut volumes = HashMap::new();
+        let mut volumes = BTreeMap::new();
         volumes.insert("one-from".to_owned(), "one-dest".to_owned());
         volumes.insert("two-from".to_owned(), "two-dest".to_owned());
 
-        let mut env_vars = HashMap::new();
+        let mut env_vars = BTreeMap::new();
         env_vars.insert("one-key".to_owned(), "one-value".to_owned());
         env_vars.insert("two-key".to_owned(), "two-value".to_owned());
 
         let image = HelloWorld { volumes, env_vars };
 
-        let mut docker = Command::new("docker");
-        let run_args = RunArgs::default();
-        let command = Client::build_run_command(&image, &mut docker, &run_args);
+        let command =
+            Client::build_run_command(&RunnableImage::from(image), Command::new("docker"));
 
         println!("Executing command: {:?}", command);
 
-        assert!(format!("{:?}", command).contains(r#"-d"#));
-        assert!(format!("{:?}", command).contains(r#"-P"#));
-        assert!(format!("{:?}", command).contains(r#""-v" "one-from:one-dest"#));
-        assert!(format!("{:?}", command).contains(r#""-v" "two-from:two-dest"#));
-        assert!(format!("{:?}", command).contains(r#""-e" "one-key=one-value""#));
-        assert!(format!("{:?}", command).contains(r#""-e" "two-key=two-value""#));
+        assert_eq!(
+            format!("{:?}", command),
+            r#""docker" "run" "-e" "one-key=one-value" "-e" "two-key=two-value" "-v" "one-from:one-dest" "-v" "two-from:two-dest" "-P" "-d" "hello-world""#
+        );
```

Comment:
Having this refactoring of switching to a single assert in the tests in a separate patch would be amazing but I am also okay if you don't want to bother :)